promotion/datastore.go

```diff
@@ -153,6 +157,68 @@ func NewPostgres(databaseURL string, performMigration bool) (*Postgres, error) {
 	return pg, nil
 }
 
+// CreateOrder given the promotion type, initial number of grants and the desired value of those grants
+func (pg *Postgres) CreateOrder(totalPrice decimal.Decimal, merchantID string, status string, orderItems []OrderItem) (*Order, error) {
+	tx := pg.DB.MustBegin()
+
+	var orderID uuid.UUID
+	err := tx.Get(&orderID,
+		`WITH new_order AS (
+			INSERT INTO orders (total_price, merchant_id, status)
+			VALUES ($1, $2, $3)
+			RETURNING id
+		) SELECT id from new_order;`,
+		totalPrice, merchantID, status)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < len(orderItems); i++ {
+		orderItem := orderItems[i]
+		orderItem.OrderID = orderID
+
+		_, err = tx.NamedExec(`
+			INSERT INTO order_items (order_id, quantity, price, currency, subtotal)
+			VALUES (:order_id, :quantity, :price, :currency, :subtotal)
+		`, orderItem)
+		if err != nil {
+			return nil, err
+		}
+	}
+	err = tx.Commit()
+	if err != nil {
+		return nil, err
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return pg.GetOrder(orderID)
```

Comment:
if we can easily keep the order info from `returning *` and we already have the list why do we need to refetch from db?

Reply:
Good point - just `returned *`

https://github.com/brave-intl/bat-go/pull/226/commits/5929f3293bf1a1da83f7325e2f830bc422975ce9